Source/ITab_Lock.cs

```diff
@@ -222,5 +221,45 @@ private void OwnerCheckbox(Rect rect, Pawn pawn)
             GUI.DrawTexture(position, image);
             Text.Anchor = anchor;
         }
+
+        private static void CheckboxLabeled(Listing_Standard listing, ref bool anythingChanged, string label, ref bool checkOn, string tooltip = null, float height = 0.0f, float labelPct = 1f)
+        {
+            var current = checkOn;
+            listing.CheckboxLabeled(label, ref checkOn, tooltip, height, labelPct);
+            if (current != checkOn)
+                anythingChanged = true;
+        }
+
+        [SyncMethod(SyncContext.MapSelected)]
+        private static void SetWantedStateData(LockState newState)
```

Comment:
I believe there is a little misunderstanding. Most of my comments were to understand what drives those changes and only issue with list required change. I'm pretty sure without that knowledge, I would refactor to use getter in the future. 😅 However, I liked more your first approach with arguments rather than selector in method. Could you bring it back for `SetWantedStateData` and `SetOwnerWantedState`? Sorry for the fuss!